### ISSUE TITLE
Wrapping NamedTuple in struct is no longer bitstype in 1.7

### ISSUE THREAD

**Issue by cortner** [JuliaLang/julia#43411]
cf brief discussion on [discourse](https://discourse.julialang.org/t/struct-wrapping-namedtuple-is-not-isbits-depending-on-how-it-is-defined/72965/9) 

This bug(?) occurs on 1.7 but not on 1.6: 

```Julia
struct Wrapper{SYMS, TT}
   x::NamedTuple{SYMS, TT}
end
randnt() = Wrapper((a = randn(), b = rand(), c = rand(1:5), d = randn(SVector{3, Float64})))
X = randnt()
@allocated randnt()  # 64
isbits(X)   # false
isbits(X.x)   # true 
```

But if the wrapper is defined differently then the `isbits == true` is recovered:
```Julia
struct Wrapper1{NT} 
   x::NT
end
randnt1() = Wrapper1((a = randn(), b = rand(), c = rand(1:5), d = randn(SVector{3, Float64})))
X1 = randnt1()
@allocated randnt1() # 0 
isbits(X1) # true
isbits(X1.x) # true
```
Because of this, it is easy enough to produce a work-around, but it feels like that shouldn't be necessary and the behaviour described above should not occur?

```
julia> versioninfo()
Julia Version 1.7.0
Commit 3bf9d17731 (2021-11-30 12:12 UTC)
Platform Info:
  OS: macOS (arm64-apple-darwin21.1.0)
  CPU: Apple M1 Max
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-12.0.1 (ORCJIT, cyclone)
```
(built from source)

**Comment by maleadt** [comment]
Bisect points to https://github.com/JuliaLang/julia/pull/40947 (cc @vtjnash)

**Comment by cortner** [comment]
I'm surprised this is for 1.8? It causes pretty significant performance regressions. Yes there are workarounds, but that's a lot of work for anybody who is affected now. In my case, I maybe ended up too trusting and started building 1.7 features into my codes; I can't go back to 1.6.

**Comment by KristofferC** [comment]
1.7 is already released so I don't know what putting a 1.7 milestone would mean. Of course the fix can be backported to 1.7 when available.

**Comment by cortner** [comment]
I see so that still goes with a 1.8 label, sorry. I’ll keep an eye on that then.
